src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseUserStudyItemToStudyDTOConverter.java

```diff
@@ -37,17 +42,21 @@
     @Override
     public S convert(AbstractUserStudyListItem source) {
 
+        final Study study = source.getStudy();
         S studyDto = createResultObject();
 
-        StudyDTO baseObject = conversionService.convert(source.getStudy(), getDtoClass());
+        StudyDTO baseObject = conversionService.convert(study, getDtoClass());
         baseObject.setFavourite(source.getFavourite());
 
+        List<StudyDataSourceLink> links = study.getDataSources().stream().filter(ds -> DataSourceStatus.DELETED.equals(ds.getStatus())).collect(Collectors.toList());
```

Comment:
What is this logic for?

Reply:
When Study is being converted to StudyDTO in BaseStudyToCommonStudyDTOConverter its DataSourceLinks are get from repository via non-nativeQuery method StudyDataSourceLinkRepository#findByStudyId(Long, EntityGraph) and are filtered then by @Where in StudyDataSourceLink. So I get DataSourceLinks from incoming study (there are "deleted" among them) and convert them here.

Reply:
So `baseObject.setDataSources(dataSourceDTOS);` will set only _deleted_ sources in the resulting DTO. W/o still active DSs. Which is not appropriate behavior